src/screens/Events/EventsMap.js

```diff
@@ -104,6 +105,21 @@ class EventsMap extends Component {
     );
   }
 
+  async initializeState() {
+    const user = await checkForUserLogin();
+    if (user) await this.props.saveUserToken(user);
+    const deviceSettings = await deviceServices.get();
+    if (deviceSettings) {
+      // If there are settings in device storage, set them
+      if (deviceSettings.language) {
+        // Set language in react-localize-redux
+        this.props.setActiveLanguage(deviceSettings.language);
+      }
+      // Save settings in redux store
+      await this.props.saveDevice(deviceSettings);
+    }
```

Comment:
Why are we saving settings about the language and device into Redux from the EventsMap? If it's because we need to grab settings from local storage and configure the app on first load or something, is there some way we can do that more globally, so that EventsMap isn't responsible for it?

Reply:
That's a good question-- I put all of the first-load store fetching stuff in EventsMap because that's the first screen users land on, and putting stuff in App.js didn't work since that's where the framework-level stuff (Redux, React Navigation, NativeBase, react-localize-redux) hooks in, and I wasn't sure where else to put it because React Navigation is in there and the root component this logic should be in wasn't super clear. Open to suggestions!

Reply:
Gotcha! It's fine to leave it for now if we want to just keep moving, but one suggestion would be to create a new component underneath App.js wrapping everything else that just does the saving on componentDidMount and then renders children. Or a Higher Order Component that wraps EventsMap and does what I just described. 

I'm just imagining that, if we ever change the first screen that appears, then we'll have to move this logic, or duplicate it if there's a choice for what screen shows up first somehow. Like I said, not a huge deal right now, though. Perhaps this screen will always be first.

Reply:
Ok! Going to try to add a higher-order wrapper component to hook into in the next PR.

Reply:
I merged this is already, but FYI I just came across this in the React Navigation docs, which I might try: https://reactnavigation.org/docs/en/localization.html